src/modlunit/units.cpp

```diff
@@ -1186,12 +1186,10 @@ void nrnunit_dynamic_str(char* buf, const char* name, char* u1, char* u2) {
 
     double legacy = dynam_unit_mag(1, u1, u2);
     double modern = dynam_unit_mag(0, u1, u2);
-    // TODO: use %a instead of %.18g when translated mechanisms are compiled with
-    //       C++17 instead of C++14 to get an exact hex representation of a double
     sprintf(buf,
             "\n"
             "#define %s _nrnunit_%s[_nrnunit_use_legacy_]\n"
-            "static double _nrnunit_%s[2] = {%.18g, %g};\n",
+            "static double _nrnunit_%s[2] = {%a, %g};\n",
```

Comment:
Note this is not just a `std::` prefix addition.

Reply:
shall we revert and open a ticket for further exploration?

Reply:
I don't have a strong opinion; my mild preference would be to leave `%a` in now that we have C++17.